res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,718 @@
+    }
+};
+
+// START BEAT FX
+
+PioneerDDJ400.numFxSlots = 3;
+
+Object.defineProperty(PioneerDDJ400, 'selectedFxSlot', {
+    get: function() {
+        return engine.getValue('[EffectRack1_EffectUnit3]', 'focused_effect');
+    },
+    set: function(value) {
+        if (value <= 0 || value > PioneerDDJ400.numFxSlots) {
+            return;
+        }
+
+        engine.setValue('[EffectRack1_EffectUnit3]', 'focused_effect', value);
+
+        const isEffectEnabled = engine.getValue(PioneerDDJ400.selectedFxGroup, 'enabled');
+
+        PioneerDDJ400.toggleLight(LightsPioneerDDJ400.beatFx, isEffectEnabled);
+    },
+});
+
+Object.defineProperty(PioneerDDJ400, 'selectedFxGroup', {
+    get: function() {
+        return '[EffectRack1_EffectUnit3_Effect' + PioneerDDJ400.selectedFxSlot + ']';
+    },
+});
+
+PioneerDDJ400.beatFxLevelDepthRotate = function(_channel, _control, value){
+    const newVal = value === 0 ? 0 : (value / 0x7F);
+    const effectOn = engine.getValue(PioneerDDJ400.selectedFxGroup, 'enabled');
+    
+    if (effectOn) {
+        engine.setValue(PioneerDDJ400.selectedFxGroup, 'meta', newVal);
+    } else {
+        engine.setValue('[EffectRack1_EffectUnit3]', 'mix', newVal);
+    }
+};
+
+PioneerDDJ400.beatFxSelectPressed = function(_channel, _control, value){
+    engine.setValue(PioneerDDJ400.selectedFxGroup, 'next_effect', value);
+};
+
+PioneerDDJ400.beatFxSelectShiftPressed = function(_channel, _control, value){
+    engine.setValue(PioneerDDJ400.selectedFxGroup, 'prev_effect', value);
+};
+
+PioneerDDJ400.beatFxLeftPressed = ignoreRelease(function() {
+    PioneerDDJ400.selectedFxSlot -= 1;
+});
+
+PioneerDDJ400.beatFxRightPressed = ignoreRelease(function() {
+    PioneerDDJ400.selectedFxSlot += 1;
+});
+
+PioneerDDJ400.beatFxOnOffPressed = ignoreRelease(function() {
+    const isEnabled = !engine.getValue(PioneerDDJ400.selectedFxGroup, 'enabled');
+
+    engine.setValue(PioneerDDJ400.selectedFxGroup, 'enabled', isEnabled);
+
+    PioneerDDJ400.toggleLight(LightsPioneerDDJ400.beatFx, isEnabled);
+});
+
+PioneerDDJ400.beatFxOnOffShiftPressed = ignoreRelease(function(_channel, control, value, status) {
+    for (var i = 1; i <= PioneerDDJ400.numFxSlots; i += 1) {
+        engine.setValue('[EffectRack1_EffectUnit3_Effect' + i + ']', 'enabled', 0);
+    }
+
+    PioneerDDJ400.toggleLight(LightsPioneerDDJ400.beatFx, false);
+});
+
+PioneerDDJ400.beatFxChannel = ignoreRelease(function(_channel, control, _value, _status, group) {
+    const enableChannel1 = control === 0x10 || control === 0x14;
+    const enableChannel2 = control === 0x11 || control === 0x14;
+
+    engine.setValue(group, 'group_[Channel1]_enable', enableChannel1);
+    engine.setValue(group, 'group_[Channel2]_enable', enableChannel2);
+});
+
+PioneerDDJ400.padFxBelowPressed = ignoreRelease(function(channel, control, value, status, group) {
+    const groupAbove = group.replace(/\[EffectRack1_EffectUnit(\d+)_Effect(\d+)]/, function(all, unit, effect) {
```

Comment:
`const` is not actually supported by the interpreter; that it happens to work is only a coincidence. Please change all `const`s to `var` to avoid confusion.